hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfs/hdfs.c

```diff
@@ -2555,28 +2541,26 @@ static jthrowable hadoopRzOptionsGetEnumSet(JNIEnv *env,
         goto done;
     }
     if (opts->skipChecksums) {
-        jthr = fetchEnumInstance(env, READ_OPTION,
+        jthr = fetchEnumInstance(env, HADOOP_RO,
                   "SKIP_CHECKSUMS", &enumInst);
         if (jthr) {
             goto done;
         }
-        jthr = invokeMethod(env, &jVal, STATIC, NULL,
-                "java/util/EnumSet", "of",
-                "(Ljava/lang/Enum;)Ljava/util/EnumSet;", enumInst);
+        jthr = invokeMethod(env, &jVal, STATIC, NULL, JC_ENUM_SET,
+                "of", "(Ljava/lang/Enum;)Ljava/util/EnumSet;", enumInst);
         if (jthr) {
             goto done;
         }
         enumSetObj = jVal.l;
     } else {
-        jclass clazz = (*env)->FindClass(env, READ_OPTION);
+        jclass clazz = (*env)->FindClass(env, HADOOP_RO);
         if (!clazz) {
             jthr = newRuntimeError(env, "failed "
-                    "to find class for %s", READ_OPTION);
+                    "to find class for %s", HADOOP_RO);
             goto done;
         }
-        jthr = invokeMethod(env, &jVal, STATIC, NULL,
-                "java/util/EnumSet", "noneOf",
-                "(Ljava/lang/Class;)Ljava/util/EnumSet;", clazz);
+        jthr = invokeMethod(env, &jVal, STATIC, NULL, JC_ENUM_SET,
```

Comment:
again not your bug, but jthr isn't checked

Reply:
Fixed